### ISSUE TITLE
Type checks on some ill-formed record operators crashes with an uncaught `TypeError`

### ISSUE THREAD

**Issue by shonfeder** [informalsystems/quint#1053]
I first noticed this in the simulator

```
>>> {a: 1}.field()
node:internal/readline/emitKeypressEvents:74
            throw err;
            ^

TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at fieldConstraints (/apalache/quint/quint/dist/src/types/specialConstraints.js:71:29)
    at specialConstraints (/apalache/quint/quint/dist/src/types/specialConstraints.js:36:20)
    at /apalache/quint/quint/dist/src/types/constraintGenerator.js:102:73
    at EitherConstructor.chain (/apalache/quint/quint/node_modules/@sweet-monads/either/cjs/index.js:108:16)
    at TypeInferrer.exitApp (/apalache/quint/quint/dist/src/types/constraintGenerator.js:99:35)
    at walkExpression (/apalache/quint/quint/dist/src/IRVisitor.js:291:25)
    at walkDefinition (/apalache/quint/quint/dist/src/IRVisitor.js:209:13)
    at /apalache/quint/quint/dist/src/types/inferrer.js:26:44
    at Array.forEach (<anonymous>)
    at TypeInferrer.inferTypes (/home/sf/Sync/informal-systems/apalache/quint/quint/dist/src/types/inferrer.js:25:14)

Node.js v19.6.0
```

But (as the stack trace indicates), it's a crash during typechecking:

```
$  cat t.qnt 
module t {
    def r() = Rec("a")
}

$  quint typecheck t.qnt
quint typecheck <input>

check types and effects of a Quint specification

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --out      output file (suppresses all console output)                [string]

TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

This is related to #1054, but comes specifically from the unguarded array accessing in the record operators, e.g.:

https://github.com/informalsystems/quint/blob/123a237513bbc92d7b629566a5573f3b076f66ab/quint/src/types/specialConstraints.ts#L91-L96

Which will give us an undefined value for the second set of bindings thanks to this lovely behavior in JS:

```
> const [a, b] = []
undefined
> a
undefined
> b
```

**Comment by konnov** [comment]
I agree that we should handle errors a bit more user-friendly. At least in the CLI. Perhaps, this is what we actually want to see in the plugin output.

**Comment by shonfeder** [comment]
This is fixed with #1078 

```
>>> {a:1}.field()
static analysis error: error: [QNT000] Operator expects 2 arguments but was given 1
Checking arity for application of field
Generating constraints for 1

{a:1}.field()
^^^^^^^^^^^^^

compile error: error: Not enough arguments
{a:1}.field()
```

The more general issue of uncaught exceptions is tracked in another ticket.
